### ISSUE TITLE
custAuth problem in 8.3.4

### ISSUE THREAD

**Issue by tuxtard** [SimpleServer/SimpleServer#180]
When I try to login using custAuth on my server, I get this message "you can only use custAuth login if you are in offline mod" even though custAuth is enabled and onlineMode is set to false in the config file:

http://pastebin.com/CSC0wTf0

**Comment by sk9c00** [comment]
Hmm, custAuth only works when you have pre-registered online, or with a legit nick-changer, then log on thru a hacked client or on offline mode then your username will become "Player[num]" Then you can use login...

sadimusi I would advise implementing something like AuthMe or ExAuth on Bukkit, I'm not sure if its possible though?..

**Comment by The-PC-Tech-Guy** [comment]
Yeah, it seems to only work if both the client AND server authentication is down. There were times that the server authentication system would be down, but not the client one, allowing clients to still possess the player's username.

**Comment by heapoutofspace** [comment]
To use custAuth you obviously have to enable `onlineMode` on the server. When you can't login with your client just click the _Play offline_ button and you should be able to login with custAuth. You don't have to use any hacked or otherwise modified clients if you bought the game.

**Comment by The-PC-Tech-Guy** [comment]
No, the issue is rarely with the client, but rather the server authenticating the player (server-side authentication). This can happen if the Minecraft server-side authentication is down temporarily, or if the server has trouble connecting to the servers (due to overload, etc.)

**Comment by The-PC-Tech-Guy** [comment]
For example, this issue happened last night - I was able to log in with my profile, but the server authentication servers were down. This is why custAuth should work, regardless of profile name.

**Comment by heapoutofspace** [comment]
I think I finally understood what you want me to do. So, when you log in with your authenticated name but the server fails to check it's authenticity you are logged in as a guest instead of being kicked with a "Bad login" explanation. Is that it?

**Comment by The-PC-Tech-Guy** [comment]
Yeah, similar to that - although when the server has trouble communicating, it returns with a 502 error code (at least last night). You should be then logged in as a guest, and prompted to enter in the CustAuth password.

**Comment by dola** [comment]
If I remember correctly it should already do that... (or I did something wrong)^^

**Edit:**
ok. it doesn't, but it shouldn't be too hard to implement this.

**Comment by bryanburke** [comment]
Is this issue fixed?
